test/base.test.js

```diff
@@ -93,6 +94,30 @@ describe('lib.base', () => {
       });
     });
 
+    describe('.delete', () => {
+      const podName = 'pod-name';
+      const query = { pretty: true };
+      const body = {
+        apiVersion: 'v1',
+        kind: 'Pod',
+        propagationPolicy: 'Foreground'
+      };
+      beforeTesting('unit', () => {
+        nock(common.api.url)
+          .delete(`/api/v1/namespaces/${ common.currentName }/pods/${ podName }`, body)
+          .query(query)
+          .reply(200, { kind: 'Pod' });
+      });
+
+      only('unit', 'should bypass query string and body from arguments into request', done => {
+        common.api.ns.po.delete({ name: podName, qs: query, body: body }, (err, result) => {
+          assume(err).is.falsy();
+          assume(result).is.eql({ kind: 'Pod' });
```

Comment:
You'll need to make this more general to handle integration tests.  I suggest something simple just to ensure we're actually piping the delete response through correctly.  E.g.,

```js
assume(result.kind).is.equal('Pod');
```